src/property.ts

```diff
@@ -950,14 +952,16 @@ function Property$ensureInited(property: Property, obj: Entity): void {
 
 		// Do not initialize calculated properties. Calculated properties should be initialized using a property get rule.
 		if (!property.isCalculated) {
+			let setPendingInit = !property.isConstant;
 			Property$init(property, obj, Property$getInitialValue(property));
-			if (property.initializer)
+			if (property.initializer) {
 				obj.update(property.name, property.initializer.call(obj));
-
-			const underlyingValue = obj.__fields__[property.name];
-			// Mark the property as pending initialization if it still has no underlying value, or is the property type's default, to allow default calculation rules to run for it
-			// List properties are defaulted onInitNew instead of on access
-			if (underlyingValue === property.defaultValue && !Array.isArray(underlyingValue))
+				setPendingInit = false;
+			}
+			// Mark the property as pending initialization if the property value may need to
+			// be established by a default calculation rule, or some other external logic.
+			// This is not relevant if it is a constant or its value is established via an initilizer.
+			if (setPendingInit)
```

Comment:
Set pending init to true not based on the value, but whether or not it was initialized with an initializer.

Also, constants should not be pending init.